store/tikv/gcworker/gc_worker.go

```diff
@@ -1038,11 +1050,6 @@ func (w *GCWorker) genNextGCTask(bo *tikv.Backoffer, safePoint uint64, key kv.Ke
 func (w *GCWorker) doGC(ctx context.Context, safePoint uint64, concurrency int) error {
 	metrics.GCWorkerCounter.WithLabelValues("do_gc").Inc()
 
-	err := w.saveSafePoint(w.store.GetSafePointKV(), tikv.GcSavedSafePoint, safePoint)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
```

Comment:
@MyonKeminta Should I move this sleep after save safe point?

Reply:
What if some tidb does not update their safepoint after we sleep(gcSafePointCacheInterval)?

Reply:
Definitely no!

Reply:
@winkyao Some tidb will report "ts may fall behink safepoint". It works kind like the DDL lease.

Reply:
@tiancaiamao could you please explain what the problem is?